keyvalue/clients/src/main/java/tachyon/client/keyvalue/KeyValueIterator.java

```diff
@@ -31,10 +31,11 @@
   boolean hasNext();
 
   /**
+   * Throws a {@link java.util.NoSuchElementException} if there are no more pairs.
```

Comment:
I think we should consider thoroughly whether to throw exception /what exception to throw for this method. But that is probably a separate PR. Right now throwing a runtime exception for an expected case is improper for me.

@HorizonNet I think it is fine for this PR like your change here. I will probably submit followup PRs to address the exception issues better for this file.